js/ai/src/generate.ts

```diff
@@ -635,6 +635,9 @@ export async function generate<
   if (resolvedOptions.output?.schema || resolvedOptions.output?.jsonSchema) {
     // find a candidate with valid output schema
     const candidateErrors = response.candidates.map((c) => {
+      // only validate candidates that have output
```

Comment:
Wait... If output schema is set, doesn't that mean we should always expect the candidate to have an output matching the described schema? 

Is there a case where empty output & output=schema is considered valid?

Reply:
The scenario is when there is no `text` or `data` in the message at all, but only e.g. a `toolRequest`. It's a fair point though that an empty text or data response should still be validated, so I've changed the logic to explicitly check for the existence of at least one text or data part.